src/neptune/internal/backends/nql.py

```diff
@@ -85,3 +101,19 @@ def __str__(self) -> str:
             value = f'"{self.value}"'
 
         return f"(`{self.name}`:{self.type.value} {self.operator.value} {value})"
+
+
+@dataclass
+class RawNQLQuery(NQLQuery):
+    query: str
+
+    def eval(self) -> NQLQuery:
```

Comment:
Are you sure that we need this `eval`? Before we've been using just `__str__` for evaluation and it was enough. I just got a feeling that we've introduced some complexity without a strong reason. But to be clear -> That just a feeling and probably I'm missing something.

Reply:
Imagine you want to run `fetch_runs_table(query='', trashed=False)`. This will under the hood build query
```
NQLQueryAggregate(
    items=[
        RawNQLQuery(''),
        NQLQueryAttribute(
                name="sys/trashed", type=NQLAttributeType.BOOLEAN, operator=NQLAttributeOperator.EQUALS, value=False
            ),
    ],
    aggregator=NQLAggregator.AND,
)
```
which would give us `() AND (sys/trashed: bool = False)` after applying `str()` without any evaluation. This is invalid query.

Thats why we need a way to remove empty queries from `NQLQueryAggregate` and imo `.eval()` method of `NQLQuery` is the best place for that

Reply:
Ok